mvndz/src/main/java/com/gd/internship/me/DZ02.java

```diff
@@ -0,0 +1,133 @@
+import java.util.LinkedList;
+import java.util.List;
+import com.github.javafaker.Faker;
+import com.gd.internship.me.AlgoTime;
+
+
+public class DZ02 {
+
+
+    public static void main(String[] args) {
+
+        List<String> stringsAL = new ArrayList<>();
+        List<String> stringsLL = new LinkedList<>();
+        AlgoTime timer = new AlgoTime();
+        long timeAL, timeLL;
+
+        setValues(stringsAL);
+        setValues(stringsLL);
+
+        System.out.println(stringsAL);
+
+        //ADDING
+
+        //ArrayList
+        timer.setStartTime();
+        stringsAL.add("To the end");
+        timer.setEndTime();
+        timeAL = timer.getTotalTime();
+
+        //LinkedList
+        timer.setStartTime();
+        stringsLL.add("To the end");
+        timer.setEndTime();
+        timeLL = timer.getTotalTime();
+
+        estimateLists(timeAL,timeLL);
+
+        //ArrayList
+        timer.setStartTime();
+        stringsAL.add( 0, "To the head");
+        timer.setEndTime();
+        timeAL = timer.getTotalTime();
+
+        //LinkedList
+        timer.setStartTime();
+        stringsLL.add(0, "To the head");
+        timer.setEndTime();
+        timeLL = timer.getTotalTime();
+
+        estimateLists(timeAL,timeLL);
+
+//        //ArrayList
+        timer.setStartTime();
+        stringsAL.add( 450000, "To the middle");
+        timer.setEndTime();
+        timeAL = timer.getTotalTime();
+
+        //LinkedList
+        timer.setStartTime();
+        stringsLL.add(450000, "To the middle");
+        timer.setEndTime();
+        timeLL = timer.getTotalTime();
+
+        estimateLists(timeAL,timeLL);
+
+        // DELETING
+
+        //ArrayList
+        timer.setStartTime();
+        stringsAL.remove("To the end");
+        timer.setEndTime();
+        timeAL = timer.getTotalTime();
+
+        //LinkedList
+        timer.setStartTime();
+        stringsLL.remove("To the end");
+        timer.setEndTime();
+        timeLL = timer.getTotalTime();
+
+        estimateLists(timeAL,timeLL);
+
+        //ArrayList
+        timer.setStartTime();
+        stringsAL.remove( 0);
+        timer.setEndTime();
+        timeAL = timer.getTotalTime();
+
+        //LinkedList
+        timer.setStartTime();
+        stringsLL.remove(0);
+        timer.setEndTime();
+        timeLL = timer.getTotalTime();
+
+        estimateLists(timeAL,timeLL);
+
+        //ArrayList
+        timer.setStartTime();
+        stringsAL.remove( 450000);
+        timer.setEndTime();
+        timeAL = timer.getTotalTime();
+
+        //LinkedList
+        timer.setStartTime();
+        stringsLL.remove(450000);
+        timer.setEndTime();
+        timeLL = timer.getTotalTime();
+
+        estimateLists(timeAL,timeLL);
+    }
+
+    // sorry, but 100k it's too long for my PC
+    public static void setValues(List<String> someList){
+        for (int i = 0; i < 900000; i++){
+            Faker faker = new Faker();
+            someList.add(faker.dune().character());
```

Comment:
Seems you are getting a lot of duplicates here. I would play a bit with the population logic here e.g.
```|
...
final dice = Math.random();
if (dice > 0.7) {
     someList.add(faker.dune().character());
} else id (dice > 0.5) {
...
}
...
...
```